src/tests/ftest/util/dmg_utils.py

```diff
@@ -1341,6 +1341,35 @@ def check_system_query_status(data):
     return not bool(failed_rank_list)
 
 
+def get_json_response(test, data, key, description):
```

Comment:
More a question for @phender. Should this function use `raise` instead of `test.fail`? Then it wouldn't need the `test` argument at all

Reply:
Yes, as we discussed we should use raises instead of `Test.fail()`, but I think that's better handled in a separate PR and we should stick with the status quo for now.

Reply:
Agree